packages/shorebird_cli/lib/src/code_push_client_wrapper.dart

```diff
@@ -616,6 +617,66 @@ aar artifact already exists, continuing...''',
     return thinnedArchiveDirectory;
   }
 
+  /// Registers and uploads macOS release artifacts to the Shorebird server.
+  Future<void> createMacosReleaseArtifacts({
+    required String appId,
+    required int releaseId,
+    required String appPath,
+    required bool isCodesigned,
+    required String? podfileLockHash,
+    required String? supplementPath,
+  }) async {
+    final createArtifactProgress = logger.progress('Uploading artifacts');
+    final tempDir = await Directory.systemTemp.createTemp();
+    final zippedApp = File(p.join(tempDir.path, '${p.basename(appPath)}.zip'));
+    await ditto.archive(source: appPath, destination: zippedApp.path);
+
+    try {
+      await codePushClient.createReleaseArtifact(
+        appId: appId,
+        releaseId: releaseId,
+        artifactPath: zippedApp.path,
+        arch: 'app',
+        platform: ReleasePlatform.macos,
+        hash: sha256.convert(await zippedApp.readAsBytes()).toString(),
```

Comment:
I probably would have had a hash function somewhere rather than encoding the use of sha256 in multiple places. 🤷

Reply:
Will clean that up in a future PR